components/ActionBox/components/DealActionBox.tsx

```diff
@@ -0,0 +1,265 @@
+import Image from 'next/image';
+import styled from 'styled-components';
+import { useEffect, useState, useMemo } from 'react';
+
+import Info from 'assets/svg/info.svg';
+
+import Connector from 'containers/Connector';
+
+import { Status } from 'components/DealStatus';
+import QuestionMark from 'components/QuestionMark';
+import ConfirmTransactionModal from 'components/ConfirmTransactionModal';
+
+import { FlexDivRowCentered, Tooltip } from '../../common';
+
+import {
+	Container,
+	ErrorNote,
+	ContentContainer,
+	ActionBoxInputLabel,
+	InputContainer,
+	ActionBoxInput,
+	ActionBoxMax,
+	ActionBoxHeaderWrapper,
+	ActionBoxHeader,
+	FlexDivCenterRow,
+	ActionButton,
+} from './commons';
+
+import { TransactionType, TransactionStatus } from 'constants/transactions';
+import { statusToText, swimmingPoolID } from 'constants/pool';
+
+const DealActionBox = ({
+	poolId,
+	dealRedemptionData,
+	purchaseCurrency,
+	transaction: { txType, txState, setTxType, setGasPrice, gasLimitEstimate, onSubmit },
+	input: { placeholder, label, maxValue, inputValue, setInputValue, setIsMaxValue },
```

Comment:
lets move away from the `input: { ... }` object. It was a terrible decision that I regret making when the codebase was just getting started

Reply:
+1

Reply:
I have some ideas that I would like to talk about